paddle/fluid/operators/repeat_interleave_op.cu

```diff
@@ -0,0 +1,307 @@
+
+      PADDLE_ENFORCE_EQ(repeats_tensor->dims()[0] == in->dims()[dim], true,
+                        platform::errors::InvalidArgument(
+                            "The length of Input(RepeatsTensor) must be the "
+                            "same as length of Input(X) in axis. "
+                            "But received: [%s], required: [%d].",
+                            repeats_tensor->dims()[0], in->dims()[dim]));
+
+      const auto& index_type = repeats_tensor->type();
+      bool index_type_match = index_type == framework::proto::VarType::INT64 ||
+                              index_type == framework::proto::VarType::INT32;
+      PADDLE_ENFORCE_EQ(
+          index_type_match, true,
+          platform::errors::InvalidArgument(
+              "Input(RepeatsTensor) holds the wrong type, it holds %s, but "
+              "desires to be %s or %s",
+              paddle::framework::DataTypeToString(index_type),
+              paddle::framework::DataTypeToString(
+                  framework::proto::VarType::INT32),
+              paddle::framework::DataTypeToString(
+                  framework::proto::VarType::INT64)));
+
+      if (index_type == framework::proto::VarType::INT64) {
+        RepeatsTensor2IndexTensor<DeviceContext, int64_t>(*repeats_tensor,
+                                                          &index);
+
+        const int64_t* index_data = index.data<int64_t>();
+        auto output_dim = framework::vectorize(in->dims());
+        output_dim[dim] = index.dims()[0];
+        out->Resize(framework::make_ddim(output_dim));
+        auto* out_data = out->mutable_data<T>(context.GetPlace());
+        int64_t numel = out->numel();
+        int64_t size = output_dim[dim];
+        int64_t delta = input_dim[dim] - size;
+
+        index_select_cuda_kernel<T, int64_t><<<
+            (numel + PADDLE_CUDA_NUM_THREADS - 1) / PADDLE_CUDA_NUM_THREADS,
+            PADDLE_CUDA_NUM_THREADS, 0, stream>>>(in_data, out_data, index_data,
+                                                  numel, stride, size, delta);
+      } else {
+        RepeatsTensor2IndexTensor<DeviceContext, int>(*repeats_tensor, &index);
+
+        const int* index_data = index.data<int>();
+        auto output_dim = framework::vectorize(in->dims());
+        output_dim[dim] = index.dims()[0];
+        out->Resize(framework::make_ddim(output_dim));
+        auto* out_data = out->mutable_data<T>(context.GetPlace());
+        int64_t numel = out->numel();
+        int64_t size = output_dim[dim];
+        int64_t delta = input_dim[dim] - size;
+
+        index_select_cuda_kernel<T, int><<<
+            (numel + PADDLE_CUDA_NUM_THREADS - 1) / PADDLE_CUDA_NUM_THREADS,
+            PADDLE_CUDA_NUM_THREADS, 0, stream>>>(in_data, out_data, index_data,
+                                                  numel, stride, size, delta);
+      }
+    } else if (repeats > 0) {
+      int index_size = in->dims()[dim] * repeats;
```

Comment:
in->dims()[dim] will return data type of int64_t, index_size should be define as int64_t ?